src/main/java/app/controller/settings/Settings.java

```diff
@@ -35,12 +36,95 @@ public void addFurniture(Rectangle2D rectangle, FurnitureType type)
         furniture.add(new SettingsObject(rectangle, type));
     }
 
-    public void addTeleport(Rectangle2D rectangle, Vector vector)
+    public void addTeleport(Rectangle2D rectangle, Vector vector, double teleportRotation)
     {
-        furniture.add(new SettingsObject(rectangle, vector));
+        furniture.add(new SettingsObject(rectangle, vector, teleportRotation));
+    }
+
+    public void addTexture(Rectangle2D rectangle, Integer textureType, Integer textureOrientation)
+    {
+        furniture.add(new SettingsObject(rectangle, textureType, textureOrientation));
     }
 
     public void addSoundSource(Vector position, double amplitude){
         soundSources.add(new SoundSourceBase(position, amplitude));
     }
+
+    public SettingsObject getTargetArea() {
+        for(SettingsObject obj: furniture) {
+            if(obj.getType()==FurnitureType.TARGET)
+                return obj;
+        }
+        return null;
```

Comment:
All of these get methods do the same thing -> lots of repeated code!
Replace with something smarter like:

```java
 public List<SettingsObject> getFurniture(FurnitureType type) 
 {
      furniture.stream()
                   .filter(obj -> obj.getType() == type)
                   .toList();  
 }
```

This replaces all of the get methods in one go

Reply:
true, I didn't think of that.
I will do that